dnf/base.py

```diff
@@ -724,6 +724,7 @@ def iter_userinstalled(self):
                 if self.history.user_installed(pkg))
 
     def _run_hawkey_goal(self, goal, allow_erasing):
+        goal.protect_running_kernel = self.conf.protect_running_kernel
```

Comment:
This would ideally also go to the place where Goal is created. But I was thinking since you've added it to the functions that create Goal in `libdnf`, this wouldn't be needed at all. I tested it though and the libdnf changes don't seem to be sufficient (unless I've made a mistake somewhere). But, looking at the code paths, it should work (`dnf.goal.Goal` is a class exposed through `python/hawkey/goal-py.cpp` in `libdnf` and it does call `hy_goal_create()` in init.

Any idea why?

Reply:
It's because the value of ``protect_running_kernel`` is at that point taken from the config as it was read from the config file and setting the value from ``--setopt`` in dnf doesn't have effect on it. I think that's why it had to be put into dnf and not libdnf... But it surprised me now that it goes through this code path, too, so I am probably still missing something. I'll look into it a bit more, but I don't think it can be changed only in libdnf.